packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -80,29 +85,68 @@ class CupertinoNavigationBar extends StatelessWidget implements ObstructingPrefe
     Key key,
     this.leading,
     this.automaticallyImplyLeading = true,
+    this.automaticallyImplyMiddle = true,
+    this.previousPageTitle,
     this.middle,
     this.trailing,
     this.border = _kDefaultNavBarBorder,
     this.backgroundColor = _kDefaultNavBarBackgroundColor,
     this.padding,
     this.actionsForegroundColor = CupertinoColors.activeBlue,
   }) : assert(automaticallyImplyLeading != null),
+       assert(automaticallyImplyMiddle != null),
        super(key: key);
 
   /// Widget to place at the start of the navigation bar. Normally a back button
   /// for a normal page or a cancel button for full page dialogs.
+  ///
+  /// If null and [automaticallyImplyLeading] is true, an appropriate button
+  /// will be automatically created.
   final Widget leading;
 
   /// Controls whether we should try to imply the leading widget if null.
   ///
   /// If true and [leading] is null, automatically try to deduce what the [leading]
   /// widget should be. If [leading] widget is not null, this parameter has no effect.
   ///
+  /// Specifically [CupertinoNavigationBar] would:
+  ///
+  /// 1. Show a 'Close' button if the current route is a `fullscreenDialog`.
+  /// 2. Show a back chevron with [previousPageTitle] if [previousPageTitle] is
+  ///    not null.
+  /// 3. If the current route is a [CupertinoPageRoute] and the previous route
```

Comment:
1 and 2 have the form "show ... if", but 3 has the form "if ... show", which is weird

Reply:
changed